packages/toaster/components/useAutoDismissToasts.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from "react";
+import { uuid } from "../../utilities/dataUtilities";
+import Toast from "./Toast";
+
+type PassedToastProps = {
+  title: React.ReactElement<HTMLElement> | string;
+  description?: React.ReactElement<HTMLElement> | string;
+  appearance?: "default" | "success" | "warning" | "danger";
+};
+
+type ToastProps = PassedToastProps & {
+  id: string;
+};
+
+const useAutoDismissToasts = () => {
+  const [currentToasts, setToasts] = React.useState<ToastProps[]>([]);
+
+  const removeToast = (id: string) => {
+    setToasts(toasts => toasts.filter(toast => toast.id !== id));
+  };
+
+  const addToast = ({ title, description, appearance }: PassedToastProps) => {
+    setToasts(toasts => [
+      ...toasts,
+      { id: uuid(), title, description, appearance }
+    ]);
+  };
+
+  const toasts = currentToasts.map(toastProps => (
+    <Toast
+      autodismiss
+      id={toastProps.id}
+      key={toastProps.id}
+      title={toastProps.title}
+      description={toastProps.description}
+      appearance={toastProps.appearance}
+      onDismiss={() => removeToast(toastProps.id)}
+    />
+  ));
+
+  return [toasts, addToast] as const;
+};
+export default useAutoDismissToasts;
```

Comment:
I don't love how we abstract out the `Toast` because what if I want to configure those props a certain way.

Reply:
looking at this in code, is this providing much utility? I can see where a consumer of toaster would want to customize their state and how they handle it.

Reply:
I would say it is a convenience method, and you can configure props to an extent through the function. I am open to suggestion but I feel like we will need some sort of method to create this short lived components one way or another.